### ISSUE TITLE
Multiple directories created during upload to Google Drive

### ISSUE THREAD

**Issue by quadnone** [ncw/rclone#2485]
#### What is the problem you are having with rclone?

Continuing with https://forum.rclone.org/t/multiple-directories-created-during-upload-to-google-drive/6569 attached two scripts to illustrate the problem

Case 1: first.sh
In this script, we try to upload 8 small files to gdrive:rclonefirst/ , each rclone copy runs in the background in parallel. This script almost everytime will illustrate the multiple directories problem

Case 2: firstwork.sh
Similar to first.sh, except that a rclone mkdir is called to first create the remote directory. There is a pause of 5 seconds but not sure if its needed at all or if needed by how much. 5 seconds seems to work. With this script there is only one rclonefirst/ directory created on the remote an all files go there.

Seems to be that Google Drive does not check if a directory exists by name before creating another with the same name since internally it seems to use file identifiers. Or is this an issue with rclone itself ?

[firstwork.sh.txt](https://github.com/ncw/rclone/files/2319567/firstwork.sh.txt)
[first.sh.txt](https://github.com/ncw/rclone/files/2319569/first.sh.txt)

Usage of scripts would be
sh first.sh
sh firstwork.sh

#### What is your rclone version (eg output from `rclone -V`)
rclone v1.42
- os/arch: linux/amd64
- go version: go1.10.1

#### Which OS you are using and how many bits (eg Windows 7, 64 bit)
Linux CentOS 6.9
uname -a
Linux centos2 2.6.32-696.23.1.el6.x86_64 #1 SMP Tue Mar 13 22:44:18 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux

####  Which cloud storage system are you using? (eg Google Drive)
Google Drive

**Comment by ncw** [comment]
I think I answered this on the forum..

> Is this something which is a known issue with Google Drive ?

Yes this is a known issue with Google Drive - see https://github.com/ncw/rclone/issues/28

It is something to do with eventual consistency on drive one thread creates the directory and the other tries to read the directory, doesn’t see it because of eventual consistency and creates another.

Because of this rclone has the rclone dedupe command which will fix duplicate files and directories for you with a variety of resolution strategies.

I noticed that even Google photos on android creates duplicate images on drive so it isn’t just rclone!

I think this behaviour is worse on TeamDrives - are you using one of those?
